### PR TITLE
Clarify names in salsa example

### PR DESCRIPTION
**Pull request by ndporter**
The ingredients lists sound like actual mild and hot salsa already. By making more intuitive and informative names, we can model variable naming best practices and make it easier to follow the example. To do that, I edited both the code and the surrounding comments and text.

<details>
<summary><strong>Instructions</strong></summary>

Thanks for contributing! :heart:

If this contribution is for instructor training, please email the link to this contribution to
checkout@carpentries.org so we can record your progress. You've completed your contribution
step for instructor checkout by submitting this contribution!

Keep in mind that **lesson maintainers are volunteers** and it may take them some time to
respond to your contribution. Although not all contributions can be incorporated into the lesson
materials, we appreciate your time and effort to improve the curriculum. If you have any questions
about the lesson maintenance process or would like to volunteer your time as a contribution
reviewer, please contact The Carpentries Team at team@carpentries.org.

You may delete these instructions from your comment.

\- The Carpentries
</details>

### CODE REVIEW

File: _episodes/04-lists.md [hunk 1]
```diff
@@ -129,31 +129,31 @@ does not.
...
 >
 > ~~~
-> salsa = ['peppers', 'onions', 'cilantro', 'tomatoes']
-> my_salsa = salsa        # <-- my_salsa and salsa point to the *same* list data in memory
-> salsa[0] = 'hot peppers'
-> print('Ingredients in my salsa:', my_salsa)
+> mlld_salsa = ['peppers', 'onions', 'cilantro', 'tomatoes']
+> hot_salsa = mild_salsa        # <-- mild_salsa and hot_salsa point to the *same* list data in memory
+> hot_salsa[0] = 'hot peppers'
+> print('Ingredients in mild salsa:', mild_salsa)
```
Comment (maxim-belkin) [inline _episodes/04-lists.md:135]: Could you please make sure this and a few other lines below conform to the proposed change of (salsa ~> mild_salsa) and (my_salsa ~> hot_salsa)?

Comment (ndporter) [reply _episodes/04-lists.md:135]: I went back through and did an updated patch that I believe caught any remaining references later in the episode.

Comment (maxim-belkin) [reply _episodes/04-lists.md:135]: Thanks, @ndporter, but my point was that you proposed to change `salsa` to `mild_salsa` and `my_salsa` to `hot_salsa`, whereas this line changes `my_salsa` to `mild_salsa`.

File: _episodes/04-lists.md [hunk 2]
```diff
@@ -129,39 +129,43 @@ does not.
...
+> Ingredients in hot salsa: ['hot peppers', 'onions', 'cilantro', 'tomatoes']
 > ~~~
 > {: .output}
 >
 > Because of pitfalls like this, code which modifies data in place can be more difficult to
 > understand. However, it is often far more efficient to modify a large data structure in place
 > than to create a modified copy for every small change. You should consider both of these aspects
 > when writing your code.
-{: .callout}
+> {: .callout}
```
Comment (maxim-belkin) [inline _episodes/04-lists.md:168]: please undo this change

Comment (noatgnu) [reply _episodes/04-lists.md:168]:
```suggestion
 {: .callout}
```
